bin/utils/tanssi-utils/src/xcm_payload.rs

```diff
@@ -0,0 +1,311 @@
+
+        let destination = self.build_foreign_destination(beneficiary);
+
+        VersionedXcmMessage::V1(MessageV1 {
+            chain_id: 1,
+            command: Command::SendNativeToken {
+                token_id: H256::from(token_id),
+                destination,
+                amount: self.amount,
+                fee: self.fee,
+            },
+        })
+    }
+
+    fn build_erc20_relay(&self, beneficiary: &[u8]) -> VersionedXcmMessage {
+        let token_address = self.parse_token_address();
+        let token = H160::from_slice(&token_address[12..]);
+        let destination = self.build_account32_destination(beneficiary);
+
+        VersionedXcmMessage::V1(MessageV1 {
+            chain_id: 1,
+            command: Command::SendToken {
+                token,
+                destination,
+                amount: self.amount,
+                fee: self.fee,
+            },
+        })
+    }
+
+    fn build_erc20_container(&self, beneficiary: &[u8]) -> VersionedXcmMessage {
+        let token_address = self.parse_token_address();
+        let token = H160::from_slice(&token_address[12..]);
+        let destination = self.build_foreign_destination(beneficiary);
+
+        VersionedXcmMessage::V1(MessageV1 {
+            chain_id: 1,
+            command: Command::SendToken {
+                token,
+                destination,
+                amount: self.amount,
+                fee: self.fee,
+            },
+        })
+    }
+
+    fn decode_beneficiary(&self) -> Vec<u8> {
+        let hex_trimmed = self
+            .beneficiary
+            .strip_prefix("0x")
+            .unwrap_or(&self.beneficiary);
+        hex::decode(hex_trimmed).expect("invalid hex in beneficiary")
+    }
+
+    fn parse_token_address(&self) -> [u8; 32] {
+        let addr = self
+            .token_address
+            .as_ref()
+            .expect("token_address is required for ERC20");
+        let hex_trimmed = addr.strip_prefix("0x").unwrap_or(addr);
+        let bytes = hex::decode(hex_trimmed).expect("invalid hex for token_address");
+        let mut arr = [0u8; 32];
+        if bytes.len() == 20 {
+            arr[12..].copy_from_slice(&bytes);
+        } else if bytes.len() == 32 {
+            arr.copy_from_slice(&bytes);
+        } else {
+            panic!("token_address must be 20 or 32 bytes");
+        }
+        arr
+    }
+
+    fn parse_genesis_hash(&self) -> NetworkId {
+        if let Some(ref h) = self.genesis_hash {
+            let hex_trimmed = h.strip_prefix("0x").unwrap_or(h);
+            let bytes = hex::decode(hex_trimmed).expect("invalid hex for genesis_hash");
+            let mut genesis_array = [0u8; 32];
+            genesis_array.copy_from_slice(&bytes);
+            NetworkId::ByGenesis(genesis_array)
+        } else {
+            NetworkId::ByGenesis(DANCELIGHT_GENESIS_HASH)
+        }
+    }
+
+    fn parse_token_location(&self) -> Location {
```

Comment:
If `Location` implements the serde traits, I would simply use `serde_json::from_str` here, this way we support arbitrarily complex locations and don't need custom parsing.